imports/plugins/included/payments-stripe/client/checkout/components/injectedCardForm.js

```diff
@@ -0,0 +1,278 @@
+      incorrect_cvc: i18next.t("checkout.errorMessages.incorrectCVC"),
+      incorrect_number: i18next.t("checkout.errorMessages.incorrectNumber"),
+      incorrect_zip: i18next.t("checkout.errorMessages.incorrectZIP"),
+      invalid_cvc: i18next.t("checkout.errorMessages.invalidCVC"),
+      invalid_expiry_month: i18next.t("checkout.errorMessages.invalidExpiryMonth"),
+      invalid_expiry_year: i18next.t("checkout.errorMessages.invalidExpiryYear"),
+      invalid_expiry_month_past: i18next.t("checkout.errorMessages.incompleteExpiryMonthPast"),
+      invalid_expiry_year_past: i18next.t("checkout.errorMessages.incompleteExpiryYearPast"),
+      invalid_number: i18next.t("checkout.errorMessages.invalidNumber"),
+      postal_code_invalid: i18next.t("checkout.errorMessages.postalCodeInvalid"),
+      state_unsupported: i18next.t("checkout.errorMessages.stateUnsupported"),
+      whoops: i18next.t("checkout.errorMessages.whoops")
+    };
+  }
+
+  handleSubmit = (ev) => {
+    const { cardNumberErrorMessage, expDateErrorMessage, CVVErrorMessage, postalErrorMessage } = this.state;
+    const { stripe, cartId } = this.props;
+    const resubmitMessage = "Resubmit payment";
+    ev.preventDefault();
+
+    if (cardNumberErrorMessage || expDateErrorMessage || CVVErrorMessage || postalErrorMessage) {
+      return;
+    }
+
+    this.setState({
+      submitMessage: "Submitting...",
+      submitting: true
+    });
+    if (stripe) {
+      stripe.createToken().then((payload) => {
+        if (payload.error) {
+          this.setState({
+            errorMessage: "",
+            submitMessage: resubmitMessage,
+            submitting: false
+          });
+          // Do not duplicate field validation errors
+          if (payload.error.type !== "validation_error") {
+            this.setState({ errorMessage: payload.error.message });
+          }
+          return;
+        }
+        Meteor.apply("stripe/payment/createCharges", ["authorize", payload.token, cartId], {
+          wait: true,
+          onResultReceived: (error, result) => {
+            if (error || (result && result.error)) {
+              this.setState({
+                errorMessage: this.errorCodes[result.error.code] ? this.errorCodes[result.error.code] : result.error.whoops,
+                submitMessage: resubmitMessage,
+                submitting: false
+              });
+            } else {
+              Router.go("cart/completed", {}, {
+                _id: cartId
+              });
+            }
+          }
+        });
+        return;
+      }).catch(() => {
+        this.setState({
+          errorMessage: "Something went wrong. Please try again.",
+          submitMessage: resubmitMessage,
+          submitting: false
+        });
+        return;
+      });
+    }
+    return;
+  };
+
+  changeHasError = (change) => (change && change.error);
+
+  handleCardNumberChange = (change) => {
+    if (this.changeHasError(change)) {
```

Comment:
Can this `if el` block be refactored into a function? and used in the other event handlers?

Reply:
What happens within the `if else` are different for each function and `handlePostalChange` has an extra needed step. So, I don't think I can do what you suggest,

Reply:
I see, then this PR looks good to ship.